packages/compiler/core/types.ts

```diff
@@ -967,6 +988,7 @@ export interface UnionStatementNode extends BaseNode, DeclarationNode, TemplateD
 
 export interface UnionVariantNode extends BaseNode {
   readonly kind: SyntaxKind.UnionVariant;
+  readonly parent?: UnionStatementNode;
```

Comment:
I wonder if we should go through and add typed parents everywhere. Do we have others?

Reply:
created https://github.com/microsoft/cadl/pull/1468